src/foam/u2/EnumView.js

```diff
@@ -9,24 +9,80 @@ foam.CLASS({
   name: 'EnumView',
   extends: 'foam.u2.view.ChoiceView',
 
+  imports: [
+    'auth'
+  ],
+
   properties: [
     {
       class: 'Class',
       name: 'of',
       required: true
     },
     {
-      name: 'choices',
-      expression: function(of) {
-        return of ? of.VALUES.map(function(v) { return [ v, v.label ]; }) : [];
+       class: 'Boolean',
+       name: 'permissioned',
+       value: false
+    },
+    {
+      name: 'permissionResults',
+      expression: async function(of, permissioned) {
+        if ( of ) {
+          if ( permissioned ) {
+            var results = [];
+            var model = of.name.toLowerCase();
+            for ( var i = 0; i < of.VALUES.length; i++ ) {
```

Comment:
spaces fore ;'s

Reply:
Was just following ESLint, do we usually have spaces before semicolons in for loops within foam2?